src/modules/QtQuick/TextEdit.js

```diff
@@ -1,83 +1,68 @@
+    length: 'int',
+    lineCount: 'int',
+    mouseSelectionMode: 'enum',
+    persistentSelection: 'bool',
+    readOnly: 'bool',
+    renderType: 'enum',
+    selectByKeyboard: { type: 'bool', initialValue: true },
+    selectByMouse: 'bool',
+    selectedText: 'string',
+    selectedTextColor: { type: 'color', initialValue: 'yellow' },
+    selectionColor: { type: 'color', initialValue: 'pink' },
+    selectionEnd: 'int',
+    selectionStart: 'int',
+    text: 'string',
+    textDocument: 'TextDocument',
+    textFormat: 'enum',
+    textMargin: 'real',
+    verticalAlignment: 'enum',
+    wrapMode: 'enum'
+  }
+}, class {
+  constructor(meta) {
     callSuper(this, meta);
 
     var self = this;
 
-    // Properties
-    createProperty('bool', this, 'activeFocusOnPress');
-    createProperty('url', this, 'baseUrl');
-    createProperty('bool', this, 'canPaste');
-    createProperty('bool', this, 'canRedo');
-    createProperty('bool', this, 'canUndo');
-    createProperty('color', this, 'color');
-    createProperty('real', this, 'contentHeight');
-    createProperty('real', this, 'contentWidth');
-    createProperty('Component', this, 'cursorDelegate');
-    createProperty('int', this, 'cursorPosition');
-    createProperty('rectangle', this, 'cursorRectangle');
-    createProperty('bool', this, 'cursorVisible');
-    createProperty('enum', this, 'effectiveHorizontalAlignment');
-    createProperty('enum', this, 'horizontalAlignment');
-    createProperty('string', this, 'hoveredLink');
-    createProperty('bool', this, 'inputMethodComposing');
-    createProperty('enum', this, 'inputMethodHints');
-    createProperty('int', this, 'length');
-    createProperty('int', this, 'lineCount');
-    createProperty('enum', this, 'mouseSelectionMode');
-    createProperty('bool', this, 'persistentSelection');
-    createProperty('bool', this, 'readOnly');
-    createProperty('enum', this, 'renderType');
-    createProperty('bool', this, 'selectByKeyboard');
-    createProperty('bool', this, 'selectByMouse');
-    createProperty('string', this, 'selectedText');
-    createProperty('color', this, 'selectedTextColor');
-    createProperty('color', this, 'selectionColor');
-    createProperty('int', this, 'selectionEnd');
-    createProperty('int', this, 'selectionStart');
-    createProperty('string', this, 'text');
-    createProperty('TextDocument', this, 'textDocument');
-    createProperty('enum', this, 'textFormat');
-    createProperty('real', this, 'textMargin');
-    createProperty('enum', this, 'verticalAlignment');
-    createProperty('enum', this, 'wrapMode');
-
     const QMLFont = getConstructor('QtQuick', '2.0', 'Font');
     this.font = new QMLFont(this);
 
-    this.activeFocusOnPress = true;
     this.baseUrl = undefined;
-    this.canPaste = false;
-    this.canRedo = false;
-    this.canUndo = false;
-    this.color = 'white';
-    this.contentHeight = 0;
-    this.contentWidth = 0;
     this.cursorDelegate = undefined;
```

Comment:
You mean the `undefined` variables? Will take a look at that, I don't remember atm why I omitted those.